source/load-balancers/load-balancers.rst

```diff
@@ -0,0 +1,531 @@
+:code:`CommandFailedEvent`. The driver MAY decide how to expose this information.
+Drivers that have a :code:`ConnectionId` object for example, MAY choose to provide a
+:code:`serviceId` in that object. The :code:`serviceId` field is only present when
+in load balancer mode and connected to a service that is behind a load balancer.
+
+Additionally the :code:`PoolClearedEvent` MUST also contain a :code:`serviceId`
+field.
+
+
+Downstream Visible Behavioral Changes
+-------------------------------------
+
+Services MAY add a command line option or other configuration parameter, that tells the service
+it is running behind a load balancer. Services MAY also dynamically determine whether they are
+behind a load balancer.
+
+All services which terminate TLS MUST be configured to return a TLS certificate for a hostname
+which matches the hostname the client is connecting to.
+
+All services behind a load balancer that have been started with the aforementioned option MUST
+add a top level :code:`serviceId` field to their response to the :code:`hello` command. This field
+MUST be a BSON :code:`ObjectId` and SHOULD NOT change while the service is running.
+When a driver is configured to not be in load balanced mode and the service is configured behind
+a load balancer, the service MAY return an error from the driver's :code:`hello` command that
+the driver is not configured to use it properly.
+
+All services that have the behaviour of reaping idle cursors after a specified period of time MAY
+also close the connection associated with the cursor when the cursor is reaped. Conversely, those
+services MAY reap a cursor when the connection associated with the cursor is closed.
+
+All services that have the behaviour of reaping idle transactions after a specified period of time
+MAY also close the connection associated with the transaction when the transaction is reaped.
+Conversely, those services must abort a transaction when the connection associated with the
+transaction is closed.
+
+Any applications that connect directly to services and not through the load balancer MUST also
+supply the :code:`loadBalanced=true` option to the driver they use to connect.
+
+
+Q&A
+---
+
+Why use a connection string option instead of a new URI scheme?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Use of a connection string option would allow the driver to continue to use SRV records that
+pointed at a load balancer instead of a replica set without needing to change the URI provided
+to the :code:`MongoClient`. The SRV records could also provide the default :code:`loadBalanced=true`
+in the TXT records.
+
+Why explicitly opt-in to this behaviour instead of letting mongos inform the driver of the load balancer?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This was a case of complexity in implementation and the number of affected services, including
```

Comment:
This paragraph is saying that there would be changes to start-up params and `hello` responses for upstream services, but we ended up doing this anyway. Can we remove this paragraph? The second paragraph is really the important bit: it adds a lot of complexity to the driver to start in one mode and then have to transition into LB mode, so we require the user to configure the driver appropriately and just start in LB mode from the beginning.

Reply:
Removed.